File: TN-POS-SYS/TN-SYS-WEB/src/features/auth/services/s_api_auth_adm_usr.ts
// 🇻🇳 API quản trị người dùng
// 🇺🇸 User management API
import { Api_Client } from '../../../core/api/api_client';
import type { M_Tb_Auth_Usr } from '../data/models';

export class S_Api_Auth_Adm_Usr {
  private apiClient: Api_Client;

  constructor(apiClient?: Api_Client) {
    this.apiClient = apiClient || new Api_Client();
  }

  // 🇻🇳 Lấy danh sách người dùng
  // 🇺🇸 Get user list
  async list(): Promise<M_Tb_Auth_Usr[]> {
    const response = await this.apiClient.get<M_Tb_Auth_Usr[]>('/auth/admin/users');
    return response.data || [];
  }

  // 🇻🇳 Xóa người dùng
  // 🇺🇸 Delete user
  async delete(id: string): Promise<boolean> {
    const response = await this.apiClient.delete<boolean>(`/auth/admin/users/${id}`);
    return response.data ?? false;
  }

  // 🇻🇳 Thêm/Sửa người dùng
  // 🇺🇸 Upsert user
  async upsert(usr: M_Tb_Auth_Usr): Promise<string> {
    const response = await this.apiClient.post<{ id: string }>('/auth/admin/users', usr);
    return response.data?.id || '';
  }
}

File: TN-POS-SYS/TN-SYS-FLU/lib/src/features/auth/services/s_api_auth_adm_usr.dart
// 🇻🇳 API quản trị người dùng
// 🇺🇸 User management API
import '../../../core/api/api_client.dart';
import '../data/models/m_tb_auth_usr.dart';

class S_Api_Auth_Adm_Usr {
  final Api_Client apiClient;

  S_Api_Auth_Adm_Usr({Api_Client? apiClient})
      : apiClient = apiClient ?? Api_Client();

  // 🇻🇳 Lấy danh sách người dùng
  // 🇺🇸 Get user list
  Future<List<M_Tb_Auth_Usr>> list() async {
    final response = await apiClient.get<List<M_Tb_Auth_Usr>>(
      '/auth/admin/users',
      fromJson: (json) => (json as List)
          .map((e) => M_Tb_Auth_Usr.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
    return response.data ?? [];
  }

  // 🇻🇳 Xóa người dùng
  // 🇺🇸 Delete user
  Future<bool> delete(String id) async {
    final response = await apiClient.delete<bool>(
      '/auth/admin/users/$id',
      fromJson: (json) => json['data'] as bool? ?? false,
    );
    return response.data ?? false;
  }

  // 🇻🇳 Thêm/Sửa người dùng
  // 🇺🇸 Upsert user
  Future<String> upsert(M_Tb_Auth_Usr usr) async {
    final response = await apiClient.post<Map<String, dynamic>>(
      '/auth/admin/users',
      body: usr.toJson(),
      fromJson: (json) => json,
    );
    return response.data?['id'] as String? ?? '';
  }
}

